### DISCUSSION TITLE
Unable to create project

### DISCUSSION THREAD

**Discussion by IvanMtze** [microsoft/devicescript#671] [Q&A] [answered]
Hello,
I'm trying to ramp up the code base, and as part of that I'm trying to create a basic project setup. However, I'm getting the following error when creating a project using the vscode extension:

```
node:fs:2352
    return binding.writeFileUtf8(
                   ^

Error: ENOENT: no such file or directory, open '/home/{...}/{...}/project1/.yarnrc.yml'
    at writeFileSync (node:fs:2352:20)
```
Any ideas on how to solve this?

**Comment by pelikhan** [comment]
It seems that some part of the directory structure was not created before writing the file

Did you install the extension from the marketplace or built locally?

The extension sends a message a cli to execute the "init" command.

**Reply by IvanMtze** [reply]
Installed from the marketplace. I click on create a new project and then it asks for the project directory and name. Then it sends the init message to the console which ends in the following. I can see that the file yarnrc.yml does not exists. It should be created before manually by any chance? 
```
npx --yes @devicescript/cli@latest init --quiet
Configuring DeviceScript project...
node:fs:2352
    return binding.writeFileUtf8(
                   ^

Error: ENOENT: no such file or directory, open '{..})/project/.yarnrc.yml'
    at writeFileSync (node:fs:2352:20)
    at /{..})
    at Array.forEach (<anonymous>)
    at writeFiles ({..}))
    at init ({..}))
    at Command3.<anonymous> ({..})
    at Command3.listener [as _actionHandler] ({..})
    at {..})
    at Command3._chainOrCall (/{..})
    at Command3._parseCommand ({..}))
    at {..})
    at Command3._chainOrCall ({..}))
    at Command3._dispatchSubcommand ({..}))
    at Command3._parseCommand ({..})
    at Command3.parse ({..}))
    at Object.mainCli ({..}) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/{..})/project/.yarnrc.yml'
}

Node.js v21.3.0
```

**Comment by pelikhan** [comment]
Could you try to pick an existing empty folder?

What OS are running on?
________________________________

**Reply by IvanMtze** [reply]
I picked an existing empty folder, I can see the following files after the "create project" is executed are the shown in the following image:

![image](https://github.com/microsoft/devicescript/assets/48934840/5325e049-fff7-40a3-a353-46fae7d12202)

I'm using linux mint 21.2

**Reply by pelikhan** [reply]
Try upgrading to devicescript 2.15.23 and enable the "DeviceScript -> Verbose" user configuration to get more output.
You could also try to run the command line in an empty folder:

    npx --yes @devicescript/cli@latest init --verbose
    
 This is strange because it seems to write a few files before failing on ".yarn..."

**Reply by IvanMtze** [reply]
So I tested on a windows machine and it works as expected. However I'm still unable to make it work on my linux one. Here is the output of the command:
```

verbose level: 1
using devs: v2.15.23, runtime: v2.15.23, node: v21.3.0 from /{..}/.npm/_npx/5d8d10ebdfb5437c/node_modules/@devicescript/cli/built
Configuring DeviceScript project...
write /{..}/sampledevs/.devcontainer/devcontainer.json
write /{..}/sampledevs/src/tsconfig.json
write /{..}/sampledevs/.prettierrc
write /{..}/sampledevs/.vscode/extensions.json
write /{..}/sampledevs/.vscode/launch.json
write /{..}/sampledevs/.yarnrc.yml
node:fs:2352
    return binding.writeFileUtf8(
                   ^

Error: ENOENT: no such file or directory, open '{..}/sampledevs/.yarnrc.yml'
```

**Reply by IvanMtze** [reply, marked as answer]
I downgraded my Node.js version from 21.* to 20.10.* and is working now. Not sure why it was happening but I'm able to make it work now.

**Reply by pelikhan** [reply]
I test a PR for 20 and it looks ok. In general, I don't think we plan to support beyond the current LTS

**Comment by pelikhan** [comment]
[surprised]     Peli de Halleux reacted to your message:
________________________________
